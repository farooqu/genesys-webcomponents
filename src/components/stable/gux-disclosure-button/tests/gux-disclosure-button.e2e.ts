import { E2EPage, newE2EPage } from '@stencil/core/testing';
import { a11yCheck } from '../../../../../tests/e2eTestUtils';

async function newNonrandomE2EPage({
  html
}: {
  html: string;
}): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(html);
  await page.waitForChanges();
  await page.addScriptTag({
    path: 'node_modules/axe-core/axe.min.js'
  });
  await page.waitForChanges();

  return page;
}

describe('gux-disclosure-button', () => {
  it('should build', async () => {
    const page = await newNonrandomE2EPage({
      html: `<gux-disclosure-button lang="en"></gux-disclosure-button>`
    });
    const element = await page.find('gux-disclosure-button');

    expect(element).toHaveClass('hydrated');
  });

  describe('#render', () => {
    [
      '<gux-disclosure-button lang="en"></gux-disclosure-button>',
      '<gux-disclosure-button lang="en" is-open></gux-disclosure-button>',
      '<gux-disclosure-button lang="en" position="left"></gux-disclosure-button>',
      '<gux-disclosure-button lang="en" is-open position="left"></gux-disclosure-button>',
      '<gux-disclosure-button lang="en" position="right"></gux-disclosure-button>',
      '<gux-disclosure-button lang="en" is-open position="right"></gux-disclosure-button>',
      '<gux-disclosure-button lang="en" label="More Info"></gux-disclosure-button>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newNonrandomE2EPage({ html });
        const element = await page.find('gux-disclosure-button');
        const elementShadowDom = await element.find(
          'pierce/.gux-disclosure-button-container'
        );

        expect(element.outerHTML).toMatchSnapshot();
        expect(elementShadowDom).toMatchSnapshot();
      });
    });
  });

  describe('events', () => {
    describe('active', () => {
      it('should open when clicked', async () => {
        const page = await newNonrandomE2EPage({
          html: '<gux-disclosure-button lang="en"></gux-disclosure-button>'
        });
        const element = await page.find('gux-disclosure-button');
        const button = await element.find('pierce/.gux-disclosure-button');
        const panel = await element.find('pierce/.gux-disclosure-panel');
        const icon = await element.find('pierce/gux-icon');
        const onActive = await page.spyOnEvent('active');

        expect(await element.getProperty('isOpen')).toBe(false);
        expect(panel).not.toHaveClass('gux-active');
        expect(icon).toEqualAttribute('icon-name', 'arrow-solid-right');

        await button.click();
        await page.waitForChanges();

        expect(await element.getProperty('isOpen')).toBe(true);
        expect(panel).toHaveClass('gux-active');
        expect(icon).toEqualAttribute('icon-name', 'arrow-solid-left');
        expect(onActive).toHaveReceivedEventDetail(true);
      });
    });
  });

  describe('A11Y', () => {
    it('should be accessible when closed', async () => {
      const page = await newNonrandomE2EPage({
        html: `<gux-disclosure-button lang="en"></gux-disclosure-button>`
      });

      await a11yCheck(page);
    });

    it('should be accessible when open', async () => {
      const page = await newNonrandomE2EPage({
        html: `<gux-disclosure-button lang="en" is-open></gux-disclosure-button>`
      });

      await a11yCheck(page);
    });
  });
});
