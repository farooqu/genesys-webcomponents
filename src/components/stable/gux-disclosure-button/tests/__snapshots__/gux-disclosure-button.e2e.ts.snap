// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-disclosure-button #render should render component as expected (1) 1`] = `"<gux-disclosure-button lang=\\"en\\" class=\\"hydrated\\"></gux-disclosure-button>"`;

exports[`gux-disclosure-button #render should render component as expected (1) 2`] = `
<div class="gux-disclosure-button-container gux-left">
  <button aria-controls="gux-disclosure-button-panel-i" aria-expanded="false" aria-label="Disclosure button" class="gux-disclosure-button">
    <gux-icon class="hydrated" icon-name="arrow-solid-right"></gux-icon>
  </button>
  <div class="gux-disclosure-panel" id="gux-disclosure-button-panel-i" role="region">
    <slot name="panel-content"></slot>
  </div>
</div>
`;

exports[`gux-disclosure-button #render should render component as expected (2) 1`] = `"<gux-disclosure-button lang=\\"en\\" is-open=\\"\\" class=\\"hydrated\\"></gux-disclosure-button>"`;

exports[`gux-disclosure-button #render should render component as expected (2) 2`] = `
<div class="gux-disclosure-button-container gux-left">
  <button aria-controls="gux-disclosure-button-panel-i" aria-expanded="true" aria-label="Disclosure button" class="gux-disclosure-button">
    <gux-icon class="hydrated" icon-name="arrow-solid-left"></gux-icon>
  </button>
  <div class="gux-active gux-disclosure-panel" id="gux-disclosure-button-panel-i" role="region">
    <slot name="panel-content"></slot>
  </div>
</div>
`;

exports[`gux-disclosure-button #render should render component as expected (3) 1`] = `"<gux-disclosure-button lang=\\"en\\" position=\\"left\\" class=\\"hydrated\\"></gux-disclosure-button>"`;

exports[`gux-disclosure-button #render should render component as expected (3) 2`] = `
<div class="gux-disclosure-button-container gux-left">
  <button aria-controls="gux-disclosure-button-panel-i" aria-expanded="false" aria-label="Disclosure button" class="gux-disclosure-button">
    <gux-icon class="hydrated" icon-name="arrow-solid-right"></gux-icon>
  </button>
  <div class="gux-disclosure-panel" id="gux-disclosure-button-panel-i" role="region">
    <slot name="panel-content"></slot>
  </div>
</div>
`;

exports[`gux-disclosure-button #render should render component as expected (4) 1`] = `"<gux-disclosure-button lang=\\"en\\" is-open=\\"\\" position=\\"left\\" class=\\"hydrated\\"></gux-disclosure-button>"`;

exports[`gux-disclosure-button #render should render component as expected (4) 2`] = `
<div class="gux-disclosure-button-container gux-left">
  <button aria-controls="gux-disclosure-button-panel-i" aria-expanded="true" aria-label="Disclosure button" class="gux-disclosure-button">
    <gux-icon class="hydrated" icon-name="arrow-solid-left"></gux-icon>
  </button>
  <div class="gux-active gux-disclosure-panel" id="gux-disclosure-button-panel-i" role="region">
    <slot name="panel-content"></slot>
  </div>
</div>
`;

exports[`gux-disclosure-button #render should render component as expected (5) 1`] = `"<gux-disclosure-button lang=\\"en\\" position=\\"right\\" class=\\"hydrated\\"></gux-disclosure-button>"`;

exports[`gux-disclosure-button #render should render component as expected (5) 2`] = `
<div class="gux-disclosure-button-container gux-right">
  <button aria-controls="gux-disclosure-button-panel-i" aria-expanded="false" aria-label="Disclosure button" class="gux-disclosure-button">
    <gux-icon class="hydrated" icon-name="arrow-solid-left"></gux-icon>
  </button>
  <div class="gux-disclosure-panel" id="gux-disclosure-button-panel-i" role="region">
    <slot name="panel-content"></slot>
  </div>
</div>
`;

exports[`gux-disclosure-button #render should render component as expected (6) 1`] = `"<gux-disclosure-button lang=\\"en\\" is-open=\\"\\" position=\\"right\\" class=\\"hydrated\\"></gux-disclosure-button>"`;

exports[`gux-disclosure-button #render should render component as expected (6) 2`] = `
<div class="gux-disclosure-button-container gux-right">
  <button aria-controls="gux-disclosure-button-panel-i" aria-expanded="true" aria-label="Disclosure button" class="gux-disclosure-button">
    <gux-icon class="hydrated" icon-name="arrow-solid-right"></gux-icon>
  </button>
  <div class="gux-active gux-disclosure-panel" id="gux-disclosure-button-panel-i" role="region">
    <slot name="panel-content"></slot>
  </div>
</div>
`;

exports[`gux-disclosure-button #render should render component as expected (7) 1`] = `"<gux-disclosure-button lang=\\"en\\" label=\\"More Info\\" class=\\"hydrated\\"></gux-disclosure-button>"`;

exports[`gux-disclosure-button #render should render component as expected (7) 2`] = `
<div class="gux-disclosure-button-container gux-left">
  <button aria-controls="gux-disclosure-button-panel-i" aria-expanded="false" aria-label="More Info" class="gux-disclosure-button">
    <gux-icon class="hydrated" icon-name="arrow-solid-right"></gux-icon>
  </button>
  <div class="gux-disclosure-panel" id="gux-disclosure-button-panel-i" role="region">
    <slot name="panel-content"></slot>
  </div>
</div>
`;
